### ISSUE TITLE
Can't specify field with type parameter

### ISSUE THREAD

**Issue by luqmana** [PoiScript/strong-xml#7]
The derive logic seems to choke if the type of a field includes a type parameter:

```rust
use chrono::{DateTime, Utc};
use::strong_xml::XmlRead;

#[derive(XmlRead)]
#[xml(tag = "document")]
struct Document {
    #[xml(attr = "datetime")]
    datetime: DateTime<Utc>,
}
```

Gives the error:

```
error[E0107]: wrong number of type arguments: expected 1, found 0
  --> src/lib.rs:8:15
   |
8  |     datetime: DateTime<Utc>,
   |               ^^^^^^^^ expected 1 type argument

```

Workaround is just using a type alias:

```rust
use chrono::{DateTime, Utc};
use::strong_xml::XmlRead;

type UtcDateTime = DateTime<Utc>;

#[derive(XmlRead)]
#[xml(tag = "document")]
struct Document {
    #[xml(attr = "datetime")]
    datetime: UtcDateTime,
}
```

**Comment by PoiScript** [comment]
strong-xml will try to remove lifetime parameters from the input type because these parameters will make the complier confused. For example, `Foo<'bar>::from_reader` won't compile but `Foo::from_reader` will because `'bar` is undeclared in the impl block.

But this will also remove the type parameter by mistake. So the solution should be using an anonymous lifetime here, like `Foo<'_>::from_reader`, which preserves the type parameters.

This solution should works and I'll implement it tomorrow.

**Comment by luqmana** [comment]
Thanks!

**Comment by PoiScript** [comment]
Ok, it's a little embarrassing now. `Foo<'_>::from_reader` doesn't work either, you have to insert two more colons between the ident and the angle bracket: `Foo::<'_>::from_reader`. I think these colons is required to avoid grammar ambiguity.

So an improved solution is providing a trait `XmlRead` and using the ufcs: `<Foo<'_> as XmlRead>::from_reader`.

**Comment by luqmana** [comment]
@PoiScript Speaking of XmlRead, I was wondering why that and XmlWrite weren't just traits instead of the derive just adding specific methods to each type

**Comment by PoiScript** [comment]
Yes, `XmlRead` and `XmlWrite` should be a trait, just like serde do. But `XmlRead` trait requires some extra work when implementing. Assume we're going to impl `XmlRead` trait for struct `Foo<'a, 'b>`, we must make sure that the input str holds a lifetime `'c` which outlives both `'a` and `'b`: `'c: 'a + 'b`. I keep forgetting to implement that.

**Comment by PoiScript** [comment]
Fixed in v0.5.0 =)

**Comment by luqmana** [comment]
Thanks for the quick fix!
